src/main/java/org/apache/activemq/broker/scheduler/SchedulerUtils.java

```diff
@@ -93,8 +95,22 @@ public Message createMessage(long start, long end) throws Exception {
 		return message;
 	}
 
-	public Message toMessage(String id, ByteSequence payload) throws Exception {
+	public ByteSequence toByteSequence(Message message) throws Exception {
+		return wireFormat.marshal(message);
+	}
+	public Message toMessage(ByteSequence payload) throws Exception {
 		Message message = (Message)wireFormat.unmarshal(payload);
+
+		return message;
+	}
+	public Message copyProperties(Message toMessageJob, Message fromMessage) throws Exception {
+		for (Map.Entry<String,Object> entry : fromMessage.getProperties().entrySet()) {
+			toMessageJob.setProperty(entry.getKey(), entry.getValue());
+		}
+		return toMessageJob;
```

Comment:
We should be able to specify command header prefixes in case we want to separate scheduled message from removal request. It could look something like this:
`toMessageJob.setProperty(commandPrefix+entry.getKey(), entry.getValue());`
Where by default it could be left as blank, in case override is desired.